scio-test/parquet/src/main/scala/com/spotify/scio/testing/parquet/ParquetTestUtils.scala

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2024 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.testing.parquet
+
+import com.spotify.parquet.tensorflow.{
+  TensorflowExampleParquetReader,
+  TensorflowExampleParquetWriter,
+  TensorflowExampleReadSupport
+}
+import _root_.magnolify.parquet.ParquetType
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.avro.{AvroParquetReader, AvroParquetWriter, AvroReadSupport}
+import org.apache.parquet.filter2.predicate.FilterPredicate
+import org.apache.parquet.hadoop.{ParquetInputFormat, ParquetReader, ParquetWriter}
+import org.apache.parquet.io._
+import org.tensorflow.proto.example.Example
+import org.tensorflow.metadata.{v0 => tfmd}
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+object ParquetTestUtils {
+  case class ParquetMagnolifyHelpers[T: ParquetType] private[testing] (records: Iterable[T]) {
+    def withFilter(filter: FilterPredicate): Iterable[T] = {
```

Comment:
shouldn't the `ParquetType` be expected here instead?

Reply:
you mean on the method rather than on the class declaration? why?